python/GafferSceneUI/ShaderQueryUI.py

```diff
@@ -424,6 +486,30 @@ def __menuDefinition( self ) :
 
 		return result
 
+	def __browseAffectedShaders( self ) :
+
+		paths = _pathsFromAffected( self )
+
+		shaderAttributes = _shaderAttributes( self, paths, affectedOnly = True )
+
+		shaderNetworks = set().union( *[ set( a.values() ) for a in shaderAttributes.values() ] )
+
+		parentWindow = self.ancestor( GafferUI.ScriptWindow )
+		browser = GafferSceneUI.ShaderUI._ShaderParameterDialogue( shaderNetworks, "Affected Shaders", parentWindow )
+
+		queries = browser.waitForParameters( parentWindow = parentWindow )
```

Comment:
Passing `parentWindow` twice seems redundant. It's a bit off, but I think the way our other dialogues work is that you just pass the parent to `waitFor()` and don't pass it to the constructor.

Reply:
Fixed in d69f0ec2aaea35ed6bebcdf63c77a7b8020262d7

Reply:
_ShaderParameterDialogue still has `self.__parentWindow`, which I think should be redundant now?

Reply:
Fixed in b4f853d7c888808d118b0eb1afb139a3e19c3ede